### PR TITLE
Adding features to add allowed scopes, claims and grant types for Cli…

### PR DESCRIPTION
**Pull request by unnarb**
# Adding features to add allowed scopes, claims and grant types for Clients to Admin API

Attach a link to issue if relevant

## What

Adding features to add allowed scopes, claims and grant types for Clients to Admin API for ADMIN UI

## Why

Adding features to add allowed scopes, claims and grant types for Clients to Admin API for ADMIN UI

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by stullifugl — APPROVED** [review]

File: apps/services/auth-admin-api/src/app/modules/clients/client-allowed-scope.controller.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+@ApiOAuth2(['@identityserver.api/read'])
+// TODO: ADD guards when functional
+// @UseGuards(AuthGuard('jwt'))
+@ApiTags('client-allowed-scope')
+@Controller('client-allowed-scope')
+export class ClientAllowedScopeController {
+  constructor(private readonly clientsService: ClientsService) {}
+
+  /** Adds new claim to client */
```
Comment (valurefugl) [inline apps/services/auth-admin-api/src/app/modules/clients/client-allowed-scope.controller.ts:24]: /** Adds new scope..

File: apps/services/auth-admin-api/src/app/modules/clients/client-allowed-scope.controller.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  Post,
+} from '@nestjs/common'
+import { ApiCreatedResponse, ApiOAuth2, ApiTags } from '@nestjs/swagger'
+
+@ApiOAuth2(['@identityserver.api/read'])
+// TODO: ADD guards when functional
+// @UseGuards(AuthGuard('jwt'))
+@ApiTags('client-allowed-scope')
+@Controller('client-allowed-scope')
+export class ClientAllowedScopeController {
```
Comment (valurefugl) [inline apps/services/auth-admin-api/src/app/modules/clients/client-allowed-scope.controller.ts:21]: Plural? ClientAllowedScopesController

File: libs/auth-api-lib/src/lib/entities/dto/client-allowed-scope.dto.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { IsNotEmpty, IsString } from 'class-validator'
+import { ApiProperty } from '@nestjs/swagger'
+
+export class ClientAllowedScopeDTO {
```
Comment (valurefugl) [inline libs/auth-api-lib/src/lib/entities/dto/client-allowed-scope.dto.ts:4]: Suggestion: Consistency in naming, Dto vs DTO

File: libs/auth-api-lib/src/lib/services/clients.service.ts [hunk 4]
```diff
@@ -29,6 +32,12 @@ export class ClientsService {
     private clientAllowedCorsOrigin: typeof ClientAllowedCorsOrigin,
```
Comment (valurefugl) [inline libs/auth-api-lib/src/lib/services/clients.service.ts:32]: Is ClientsService getting to "fat"? Maybe split it up?

File: libs/auth-api-lib/src/lib/services/clients.service.ts [hunk 5]
```diff
@@ -250,4 +259,103 @@ export class ClientsService {
       where: { clientId: clientId, redirectUri: redirectUri },
     })
   }
+
+  /** Adds a grant type to client */
+  async addGrantType(
+    grantTypeObj: ClientGrantTypeDTO,
```
Comment (valurefugl) [inline libs/auth-api-lib/src/lib/services/clients.service.ts:265]: grantTypeObj -> clientGrantType?

**Review by valurefugl — APPROVED** [review]
Looks good, minor suggestions - mainly naming stuff.

**Pull request merged**